### ISSUE TITLE
WIP: Proposal: Sybil Attack Prevention

### ISSUE THREAD

**Issue by blrhc** [Joystream/joystream#117]
# WIP
### Background
The council has suffered from what appear to be Sybil attacks from one or more actors when we've held elections. Sybil attacks will be one of the defining challenges facing Joystream, so exploring how to tackle the issue in one area of Joystream (the faucet) would be a good place to start learning about effective ways of tackling the problem.
### Potential Solutions

- Offering considerably less from the faucet, leaving Jsgenesis team and trusted users effectively in control of elections.
- Making the faucet frictional, e.g. requiring requester to tweet something or connect to another identity (e.g. GitHub or Keybase account) when requesting tokens.
- Manually moderating requests from the faucet to detect suspicious claims.
- Removing faucet altogether and asking users to request tokens in Telegram group.

### Opportunity
As I understand it, we intend to wipe users' balances clean when Rome is launched. This would be a very good opportunity to recalibrate the system and introduce a new faucet system.

**Comment by bedeho** [comment]
Thanks for the writeup!

What is the evidence which suggests that there has been an attack? I buy it, I just wonder how you can tell.

We have started planning a new very innovative model for how to do testnets after Rome which will, even more than before, require that we deal with Sybil issues even more effectively than prior. 

Read more here

https://github.com/Joystream/joystream/issues/119

**Comment by blrhc** [comment]
In terms of evidence, it's principally that the bulk of the candidates for the most recent two or three council elections have had memos which only contained Monero addresses.

Also, the fact that so many of these "candidates" appear in such a short period of time and then suddenly no more decide to apply indicate to me that this is a human orchestrating the attack at a specific time of day within a specific time zone and then giving up or deciding to stop making any more applications.

Furthermore, in the last council election there were something like 124 votes made and they were all votes between the same network of accounts, all without memos. I don't see why anybody would decide to vote for an account with just a Monero address if it was not their own spam account designed to scoop up the Monero offered to election winners.

I am also aware that the faucet is not at all abuse-proof. Anybody with a VPN and incognito mode enabled can make unlimited claims from the faucet. There's really nothing stopping them at all apart from the captcha, and when filling in this captcha pays up to $10 per claim, it's a great deal!

**Comment by bwhm** [comment]
There are some serious whales out there, so I'm pretty sure you are ready. We could "burn" some of their holdings, but I don't really like that. They were able to find a hole in the system, so we should rather try to adapt (ref #119)

**Comment by blrhc** [comment]
I agree we need to adapt, but will this be possible before the next testnet?

Burning users tokens using the sudo key is an anti-governance position but the whales are definitely not working in the best interests of the platform and are arguably pushing out other genuine users who would also like to be council members and validators. 

These people are not coming forward and explaining that they attacked the platform in a certain way and helping us fix it. It's not testing the platform, it's abusing it.

On the other hand, I suppose we don't really want to push anyone away, as the bad actors in themselves are quite a good simulation of the threats Joystream might face on mainnet and are useful for calibration purposes.

**Comment by bwhm** [comment]
Rome will start from a fresh genesis, and partly for the reasons outlined above, we will not transfer balances (at least not in full).

My thought process was that we then let the faucet only distribute the minimum to register as a member, and little more. Requests on telegram and/or forum would be a way to get more tokens!

**Comment by blrhc** [comment]
Excellent idea!
